packages/react-instantsearch/src/core/createInstantSearchManager.js

```diff
@@ -41,6 +41,11 @@ export default function createInstantSearchManager({
     searching: false,
   });
 
+  function updateClient(client) {
+    helper.setClient(client);
+    search(); // Will it work?
```

Comment:
And I won't remove the comment until I have your opinion (because GH will hide it if there is a change at this line)